tests/test_dataset_utils.py

```diff
@@ -0,0 +1,132 @@
+                "common_elec": 6.8,
+            },
+            {
+                "county": "Denver",
+                "sector": "res",
+                "com_elec": None,
+                "res_elec": 4.2,
+                "common_elec": 5.8,
+            },
+            {
+                "county": "Adams",
+                "sector": "res",
+                "com_elec": None,
+                "res_elec": 1.3,
+                "common_elec": 4.8,
+            },
+        ]
+    )
+    records = spark.createDataFrame(
+        [
+            {"from_id": "res_elec", "to_id": "all_electricity", "from_fraction": 1.0},
+            {"from_id": "com_elec", "to_id": "all_electricity", "from_fraction": 1.0},
+            {"from_id": "common_elec", "to_id": "all_electricity", "from_fraction": 1.0},
+        ]
+    )
+    pivoted_columns = {"com_elec", "res_elec", "common_elec"}
+    yield df, records, pivoted_columns
+
+
+def test_map_and_reduce_pivoted_dimension_sum_rename(dataframes):
+    df, records, pivoted_columns = dataframes
+    operation = "sum"
+    res, new_pivoted, dropped = map_and_reduce_pivoted_dimension(
+        df, records, pivoted_columns, operation, rename=True
+    )
+    assert new_pivoted == ["all_electricity_sum"]
+    assert sorted(dropped) == sorted(pivoted_columns)
+    expected = 2.1 + 7.8 + 3.5 + 6.8 + 4.2 + 5.8 + 1.3 + 4.8
+    assert math.isclose(
+        res.select("all_electricity_sum")
+        .agg(F.sum("all_electricity_sum").alias("sum_elec"))
+        .collect()[0]
+        .sum_elec,
+        expected,
+    )
+
+
+def test_map_and_reduce_pivoted_dimension_sum_no_rename(dataframes):
+    df, records, pivoted_columns = dataframes
+    operation = "sum"
+    res, new_pivoted, dropped = map_and_reduce_pivoted_dimension(
+        df, records, pivoted_columns, operation, rename=False
+    )
+    assert new_pivoted == ["all_electricity"]
+    assert sorted(dropped) == sorted(pivoted_columns)
+    expected = 2.1 + 7.8 + 3.5 + 6.8 + 4.2 + 5.8 + 1.3 + 4.8
+    assert math.isclose(
+        res.select("all_electricity")
+        .agg(F.sum("all_electricity").alias("sum_elec"))
+        .collect()[0]
+        .sum_elec,
+        expected,
+    )
+
+
+def test_map_and_reduce_pivoted_dimension_max(dataframes):
+    df, records, pivoted_columns = dataframes
+    operation = "max"
+    res, new_pivoted, dropped = map_and_reduce_pivoted_dimension(
+        df, records, pivoted_columns, operation, rename=False
+    )
+    assert new_pivoted == ["all_electricity"]
+    assert sorted(dropped) == sorted(pivoted_columns)
+    assert res.filter("county == 'Jefferson'").collect()[0].all_electricity == 7.8
+    assert res.filter("county == 'Boulder'").collect()[0].all_electricity == 6.8
+    assert res.filter("county == 'Denver'").collect()[0].all_electricity == 5.8
+    assert res.filter("county == 'Adams'").collect()[0].all_electricity == 4.8
+
+
+def test_map_and_reduce_pivoted_dimension_min(dataframes):
```

Comment:
Do we have one for map_and_reduce_stacked_dimensions?

Reply:
We do not. We should add something similar.

Reply:
Add TODO?